File: src/pocketcloud/cloud/terminal/log/CloudLogger.php
<?php

namespace pocketcloud\cloud\terminal\log;

use pocketcloud\cloud\terminal\log\logger\Logger;

final class CloudLogger {

    private static ?Logger $logger = null;

    public static function set(?Logger $logger): void {
        self::$logger = $logger;
    }

    public static function get(): Logger {
        if (self::$logger === null) self::set(new Logger(LOG_PATH));
        return self::$logger;
    }

    public static function close(): void {
        self::$logger?->close();
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/log/CloudLogger.java
package de.pocketcloud.cloud.console.log;

import de.pocketcloud.cloud.console.log.def.MainLogger;
import de.pocketcloud.cloud.console.log.def.PrefixedLogger;

public final class CloudLogger {

    private static final de.pocketcloud.api.logging.ILogger MAIN = new MainLogger();
    private static volatile de.pocketcloud.api.logging.ILogger instance = MAIN;

    private CloudLogger() {}

    public static void set(de.pocketcloud.api.logging.ILogger logger) {
        instance = logger != null ? logger : MAIN;
    }

    public static de.pocketcloud.api.logging.ILogger get() {
        return instance;
    }

    public static PrefixedLogger prefixed(String prefix) {
        return new PrefixedLogger(get(), prefix);
    }
}